mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergFilterFactory.java

```diff
@@ -0,0 +1,156 @@
+        //https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/io/parquet/read/
+        // ParquetFilterPredicateConverter.java#L116
+        throw new UnsupportedOperationException("CONSTANT operator is not supported");
+      default:
+        throw new IllegalStateException("Unknown operator: " + tree.getOperator());
+    }
+  }
+
+  /**
+   * Translate leaf nodes from Hive operator to Iceberg operator.
+   * @param leaf Leaf node
+   * @return Expression fully translated from Hive PredicateLeaf
+   */
+  private static Expression translateLeaf(PredicateLeaf leaf) {
+    String column = leaf.getColumnName();
+    switch (leaf.getOperator()) {
+      case EQUALS:
+        return equal(column, leafToIcebergType(leaf));
+      case LESS_THAN:
+        return lessThan(column, leafToIcebergType(leaf));
+      case LESS_THAN_EQUALS:
+        return lessThanOrEqual(column, leafToIcebergType(leaf));
+      case IN:
+        return in(column, (List) leafToIcebergType(leaf));
+      case BETWEEN:
+        List<Object> icebergLiterals = leaf.getLiteralList();
+        return and(greaterThanOrEqual(column, icebergLiterals.get(0)),
+                lessThanOrEqual(column, icebergLiterals.get(1)));
+      case IS_NULL:
+        return isNull(column);
+      default:
+        throw new IllegalStateException("Unknown operator: " + leaf.getOperator());
+    }
+  }
+
+  private static Object leafToIcebergType(PredicateLeaf leaf) {
+    switch (leaf.getType()) {
+      case LONG:
+        return leaf.getLiteral() != null ? leaf.getLiteral() : leaf.getLiteralList();
+      case FLOAT:
+        return leaf.getLiteral() != null ? leaf.getLiteral() : leaf.getLiteralList();
+      case STRING:
+        return leaf.getLiteral() != null ? leaf.getLiteral() : leaf.getLiteralList();
+      case DATE:
+        //Hive converts a Date type to a Timestamp internally when retrieving literal
+        if (leaf.getLiteral() != null) {
+          return ((Timestamp) leaf.getLiteral()).toLocalDateTime().toLocalDate().toEpochDay();
+        } else {
+          //But not when retrieving the literalList
+          List<Object> icebergValues = leaf.getLiteralList();
+          icebergValues.replaceAll(value -> ((Date) value).toLocalDate().toEpochDay());
+          return icebergValues;
+        }
+      case DECIMAL:
+        if (leaf.getLiteral() != null) {
+          return BigDecimal.valueOf(((HiveDecimalWritable) leaf.getLiteral()).doubleValue());
+        } else {
+          List<Object> icebergValues = leaf.getLiteralList();
+          icebergValues.replaceAll(value -> BigDecimal.valueOf(((HiveDecimalWritable) value).doubleValue()));
+          return icebergValues;
+        }
+      case TIMESTAMP:
+        if (leaf.getLiteral() != null) {
+          Timestamp timestamp = (Timestamp) leaf.getLiteral();
+          return timestamp.toInstant().getEpochSecond() * 1000000 + timestamp.getNanos() / 1000;
```

Comment:
Not a big deal but I tend to make constants for numbers that can be misread like this, MILLION or NANOS_PER_SECOND

Reply:
A very good point, I'll update these!